packages/builder/src/builder.js

```diff
@@ -377,15 +388,13 @@ export default class Builder {
     } else if (this._nuxtPages) {
       // Use nuxt.js createRoutes bases on pages/
       const files = {}
-        ; (await glob(`${this.options.dir.pages}/**/*.{${this.supportedExtensions.join(',')}}`, {
-        cwd: this.options.srcDir,
-        ignore: this.options.ignore
-      })).forEach((f) => {
-        const key = f.replace(new RegExp(`\\.(${this.supportedExtensions.join('|')})$`), '')
-        if (/\.vue$/.test(f) || !files[key]) {
-          files[key] = f.replace(/(['"])/g, '\\$1')
+      const ext = new RegExp(`\\.(${this.supportedExtensions.join('|')})$`)
+      for (const page of await this.resolveFiles(this.options.dir.pages)) {
+        const key = page.replace(ext, '')
+        if (/\.vue$/.test(page) || !files[key]) {
```

Comment:
Change condition `||` to `&&`. See #4641

Reply:
Sorry, the problem is not in `pages` but in` layouts`

Reply:
If try register layout in module doesn't work

```js
export default {
    modules: [
        {
            handler() {
                this.addLayout('./templates/my-layout.vue', 'default');
            }
        }
    ]
}
```

The problem is here:
```js
templateVars.layouts[name] = this.relativeToBuild(
    this.options.srcDir,
    file
);
```

Reply:
@ricardogobbosouza Gotcha, I'll try to figure out a proper fix.

Reply:
@ricardogobbosouza Fix can be reviewed in #4702 😄